database/src/service/request.rs

```diff
@@ -0,0 +1,57 @@
+//! Read/write `Request`s to the database.
+//!
+//! TODO: could add `strum` derives.
+
+//---------------------------------------------------------------------------------------------------- Import
+
+//---------------------------------------------------------------------------------------------------- Constants
+
+//---------------------------------------------------------------------------------------------------- Request
+#[derive(Debug)]
+/// Either a read or write request.
+///
+/// TODO: not sure if we actually need this.
+pub enum Request {
```

Comment:
I also don't think we need this if we have separate threads for read and write ops